packages/components/fields/date-time-field/src/date-time-field.tsx

```diff
@@ -0,0 +1,238 @@
+   * <br />
+   * Unknown errors will be forwarded to `renderError`
+   */
+  errors?: TFieldErrors;
+  /**
+   * Called with custom errors. This function can return a message which will be wrapped in an ErrorMessage. It can also return null to show no error.
+   */
+  renderError?: TErrorRenderer;
+  /**
+   * Indicates if the value is required. Shows an the "required asterisk" if so.
+   */
+  isRequired?: boolean;
+  /**
+   * Indicates whether the field was touched. Errors will only be shown when the field was touched.
+   */
+  touched?: boolean;
+
+  // DateTimeInput
+  /**
+   * Used as HTML name of the input component.
+   */
+  name?: string;
+  /**
+   * Value of the input
+   */
+  value: string;
+  /**
+   * Called with an event holding the new value.
+   * <br/>
+   * Required when input is not read only. Parent should pass it back as `value`-
+   */
+  onChange: (event: TEvent) => void;
+  /**
+   * Called when input is blurred
+   */
+  onBlur?: (event: TEvent) => void;
+  /**
+   * Called when input is focused
+   */
+  onFocus?: FocusEventHandler;
+  /**
+   * Indicates that the input cannot be modified (e.g not authorized, or changes currently saving).
+   */
+  isDisabled?: boolean;
+  /**
+   * Indicates that the field is displaying read-only content
+   */
+  isReadOnly?: boolean;
+  /**
+   * Placeholder text for the input
+   */
+  placeholder?: string;
+  /**
+   * Specifies the time zone in which the calendar and selected values are shown. It also influences how entered dates and times are parsed.
+   * <br />
+   * See the `DateTimeInput` docs for more information.
+   */
+  timeZone: string;
+
+  // LabelField
+  /**
+   * Title of the label
+   */
+  title: ReactNode;
+  /**
+   * Hint for the label. Provides a supplementary but important information regarding the behaviour of the input (e.g warn about uniqueness of a field, when it can only be set once), whereas `description` can describe it in more depth. Can also receive a `hintIcon`.
+   */
+  hint?: ReactNode;
+  /**
+   * Provides a description for the title.
+   */
+  description?: ReactNode;
+  /**
+   * Function called when info button is pressed.
+   * <br />
+   * Info button will only be visible when this prop is passed.
+   */
+  onInfoButtonClick?: (
+    event: MouseEvent<HTMLButtonElement> | KeyboardEvent<HTMLButtonElement>
+  ) => void;
+  /**
+   * Icon to be displayed beside the hint text.
+   * <br />
+   * Will only get rendered when `hint` is passed as well.
+   */
+  hintIcon?: ReactElement;
+  /**
+   * Badge to be displayed beside the label.
+   * <br />
+   * Might be used to display additional information about the content of the field (E.g verified email)
+   */
+  badge?: ReactNode;
+};
+
+type TDateTimeFieldState = Pick<TDateTimeFieldProps, 'id'>;
+
+class DateTimeField extends Component<
+  TDateTimeFieldProps,
+  TDateTimeFieldState
+> {
+  static displayName = 'DateTimeField';
+
+  static defaultProps = {
```

Comment:
Just one minor remark - in your other PRs you tend to have `defaultProps` also typed

Reply:
Yes, thanks.